src/Infrastructure.VS/ThreadHandling.cs

```diff
@@ -40,8 +41,51 @@ public class ThreadHandling : IThreadHandling
 
         public async Task RunOnUIThread(Action op) => await VS.RunOnUIThread.RunAsync(op);
 
-        public T Run<T>(Func<System.Threading.Tasks.Task<T>> asyncMethod) => ThreadHelper.JoinableTaskFactory.Run(asyncMethod);
+        public T Run<T>(Func<Task<T>> asyncMethod) => ThreadHelper.JoinableTaskFactory.Run(asyncMethod);
+
+        public IAwaitableWrapper SwitchToBackgroundThread() => new TaskSchedulerAwaitableWrapper(new AwaitExtensions.TaskSchedulerAwaitable(TaskScheduler.Default));
+
+        #region Wrappers for VS TaskSchedule awaiter/awaitable structs
```

Comment:
The normal approach to making a type `awaitable` is to have two structs:
- one that acts as a factory. It implements `GetAwaiter` to return the second struct.
- a struct that has the three required methods: `OnCompleted`, `IsCompleted`, `GetResult`.